backend/src/main/java/com/crm/service/AccountServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+import com.crm.exception.user.NoSuchAccountException;
+import com.crm.model.db.AccountDetailsAdapter;
+import com.crm.model.db.AccountEntity;
+import com.crm.repository.AccountRepository;
+import com.crm.repository.RoleRepository;
+import com.crm.tools.emails.EmailGenerator;
+import com.crm.tools.emails.EmailSender;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static com.crm.exception.ErrorDict.ACCOUNT_ALREADY_EXIST;
+import static com.crm.exception.ErrorDict.ACCOUNT_DOES_NOT_EXIST;
+import static com.crm.exception.ErrorDict.ACCOUNT_IS_ACTIVE;
+
+@Service
+@RequiredArgsConstructor
+public class AccountServiceImpl implements UserDetailsService, AccountService {
+
+    private final AccountRepository accountRepository;
+    private final AccountMapper accountMapper;
+    private final EmailGenerator emailGenerator;
+    private final EmailSender emailSender;
+    private final PasswordEncoder passwordEncoder;
+    private final RoleRepository roleRepository;
+
+    @Override
+    public AccountResponse save(final NewAccountRequest newAccountRequest) {
+        AccountEntity accountEntity = setUpAccountEntity(newAccountRequest);
+
+        try {
+            findByLogin(accountEntity.getLogin());
+            if (accountEntity.getIsActivated()) {
+                throw new AccountIsActiveException(ACCOUNT_IS_ACTIVE);
+            } else {
+                throw new AccountAlreadyExistException(ACCOUNT_ALREADY_EXIST);
+            }
+        } catch (CrmLoginException e) {
+            AccountEntity account = accountRepository.save(accountEntity);
+            emailSender.sendEmail(emailGenerator.generateEmail(account));
+            return accountMapper.convertToDto(account);
+        }
+    }
+
+    private AccountEntity setUpAccountEntity(final NewAccountRequest newAccountRequest) {
+        return AccountEntity.builder()
+                .email(newAccountRequest.getEmail())
+                .login(newAccountRequest.getLogin())
+                .password(passwordEncoder.encode(newAccountRequest.getPassword()))
+                .registrationDate(newAccountRequest.getRegistrationDate())
+                .role((roleRepository.findById(2L)).orElseThrow(NoSuchElementException::new))
+                .loginAttempts(0)
+                .isActivated(false)
+                .build();
+    }
+
+    @Override
+    public AccountEntity findById(final Long id) {
+        Optional<AccountEntity> account = accountRepository.findById(id);
+        return account.orElseThrow(() -> new NoSuchAccountException(ACCOUNT_DOES_NOT_EXIST));
```

Comment:
return accountRepository.findById(id).orElseThrow(() -> new NoSuchAccountException(ACCOUNT_DOES_NOT_EXIST));

to samo dla metod poniżej